spec/policies/configuration/agent_role_policy_spec.rb

```diff
@@ -8,37 +8,48 @@
   shared_examples "permit actions" do |*actions|
     actions.each do |action|
       permissions action do
-        it { is_expected.to permit(agent_territorial_context, agent.role_in_organisation(organisation)) }
+        it { is_expected.to permit(agent_territorial_context, agent_role) }
       end
     end
   end
 
   shared_examples "not permit actions" do |*actions|
     actions.each do |action|
       permissions action do
-        it { is_expected.not_to permit(agent_territorial_context, agent.role_in_organisation(organisation)) }
+        it { is_expected.not_to permit(agent_territorial_context, agent_role) }
       end
     end
   end
 
   describe "agent with" do
     context "no admin access to this territory and no access rights" do
       let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: []) }
-      let!(:access_rights) { create(:agent_territorial_access_right, agent: agent, territory: territory) }
+      let!(:agent_role) { create(:agent_role, organisation: organisation) }
 
       it_behaves_like "not permit actions", :update?, :edit?, :create?, :destroy?
     end
 
     context "admin access to this territory" do
       let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: [territory]) }
-      let!(:access_rights) { create(:agent_territorial_access_right, agent: agent, territory: territory) }
+      let!(:agent_role) { create(:agent_role, organisation: organisation) }
 
       it_behaves_like "permit actions", :update?, :edit?, :create?, :destroy?
     end
 
+    context "admin access to a different territory" do
+      let(:agent) { create(:agent, basic_role_in_organisations: [organisation], role_in_territories: [territory]) }
+      let!(:agent_role) { create(:agent_role, organisation: create(:organisation)) }
```

Comment:
petite suggestion pour être plus clair que c’est un autre territoire éventuellement 

```suggestion
      let!(:other_territory) { create(:territory) }
      let!(:agent_role) { create(:agent_role, organisation: create(:organisation, territory: other_territory)) }
```